Services/Administration/OJS.Services.Administration.Business/Tests/Permissions/TestIdPermissionService.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace OJS.Services.Administration.Business.Tests.Permissions;
+
+using Microsoft.EntityFrameworkCore;
+using OJS.Data.Models.Tests;
+using OJS.Services.Administration.Business.Contests;
+using OJS.Services.Administration.Data;
+using OJS.Services.Common.Models.Users;
+using System.Linq;
+using System.Threading.Tasks;
+
+public class TestIdPermissionService : IEntityPermissionsService<Test, int>
+{
+    private readonly ITestsDataService testsDataService;
+    private readonly IContestsBusinessService contestsBusinessService;
+
+    public TestIdPermissionService(
+        ITestsDataService testsDataService,
+        IContestsBusinessService contestsBusinessService)
+    {
+        this.testsDataService = testsDataService;
+        this.contestsBusinessService = contestsBusinessService;
+    }
+
+    public async Task<bool> HasPermission(UserInfoModel user, int value, string action)
+    {
+        var contestId = await this.testsDataService
+            .GetByIdQuery(value)
+            .Include(t => t.Problem)
+            .ThenInclude(p => p.ProblemGroup)
+            .Select(t => t.Problem.ProblemGroup.ContestId)
+            .FirstOrDefaultAsync();
```

Comment:
Why the `Include`s?

Reply:
Because i need access to the contest id from the test

Reply:
You have access. `Include` is needed when you want to have whatever you have included in the final result from the query, but here the final result is just the `contestId`.

Reply:
Fixed.